File: c/23740.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>

typedef struct Route {
    int start;
    int end;
    int cost;
} Route;

bool contains(int value, int bottom, int top) {
    return bottom <= value && value <= top;
}

bool is_duplicate(Route* a, Route* b) {
    return contains(a->start, b->start, b->end)
        || contains(a->end, b->start, b->end)
        || contains(b->start, a->start, a->end)
        || contains(b->end, a->start, a->end);
}

int min(int a, int b) {
    return a < b ? a : b;
}

int max(int a, int b) {
    return a < b ? b : a;
}

Route* merge(Route* a, Route* b) {
    Route* result = (Route*)malloc(sizeof(Route));
    result->start = min(a->start, b->start);
    result->end = max(a->end, b->end);
    result->cost = min(a->cost, b->cost);
    return result;
}

int main() {
    int number;

    Route* routes[200001];

    scanf("%d", &number);
    for (int i = 0; i < number; i++) {
        Route* route = (Route*)malloc(sizeof(Route));
        scanf("%d %d %d", &route->start, &route->end, &route->cost);
        routes[i] = route;
    }

    int offset = 0;
    for (int i = 0; i < number - offset; i++) {
        for (int j = i + 1; j < number - offset; j++) {
            if (is_duplicate(routes[i], routes[j])) {
                routes[i] = merge(routes[i], routes[j]);
                offset++;
                for (int k = j; k < number - offset; k++) {
                    routes[k] = routes[k+1];
                }
                j--;
            }
        }
    }

    printf("%d\n", number - offset);
    for (int i = 0; i < number - offset; i++) {
        Route route = *routes[i];
        printf("%d %d %d\n", route.start, route.end, route.cost);
    }

    return 0;
}

File: python/23740.py
number = int(input())
points = set()
routes = list()
for i in range(number):
    routes.append(list(map(int, input().split())))
    points.add(routes[-1][0])
    points.add(routes[-1][1])


def routes_containing_point(point):
    result = list()
    start_min = 1000000000
    end_max = 0
    fee_min = 1000000000
    for route in routes:
        if route[0] <= point <= route[1]:
            result.append(route)
            start_min = min(route[0], start_min)
            end_max = max(route[1], end_max)
            fee_min = min(route[2], fee_min)
    return result, start_min, end_max, fee_min


results = list()
previous_intersects = list()

for point in sorted(points):
    containing_routes, start_min, end_max, fee_min = routes_containing_point(point)
    if previous_intersects and results[-1][1] >= point:
        results[-1][0] = min(start_min, results[-1][0])
        results[-1][1] = max(end_max, results[-1][1])
        results[-1][2] = min(fee_min, results[-1][2])
    else:
        results.append([start_min, end_max, fee_min])
    previous_intersects = containing_routes


print(len(results))
for result in results:
    print(result[0], result[1], result[2])

